alpa/torch/trainer.py

```diff
@@ -55,6 +56,11 @@ def compute_loss(params, bufs, inputs, targets):
                 loss_value = loss_func(out, targets)
                 return loss_value, bufs
 
+            if atorch.mode() == "dist" and isinstance(parallel_method,
+                                                      alpa.PipeshardParallel):
+                assert auto_layer_con_func is not None
+                compute_loss = auto_layer_con_func(compute_loss)
```

Comment:
This decorator function is no longer needed after this API refactor (https://github.com/alpa-projects/alpa/pull/565)